### DISCUSSION TITLE
Distribute `.duckdb` files

### DISCUSSION THREAD

**Discussion by bendnorman** [duckdb/duckdb#8162] [Q&A] [answered]
Hello! I work on an open data project called the [Public Utility Data Liberation Project](https://github.com/catalyst-cooperative/pudl) (PUDL). We clean and distribute small to medium size data via SQLite and Parquet files. Duckdb seems like a great fit for the project because it would allow us to distribute a single in-process database that can handle dozens - hundreds of gigabytes of data. **Is it possible/common for people to distribute `.duckdb` files to users or is it really intended to be used as a query tool on top of storage like parquet files?**

**Comment by yoonghm** [comment]
I think duckdb could also be distributed but you need to aware of database and binary compatibility

**Comment by wrb2** [comment, marked as answer]
DuckDB doesn't provide backward or forward compatibility of storage format, you can only read database using the exact same version as it was created with. It should 

Also, zstd compressed parquet files are smaller than DuckDB internal format, and can be read by multiple tools. You're probably better  off sticking with that.

**Reply by bendnorman** [reply]
Ah I see; thank you!

I loaded a 5 GB parquet file to a duckdb table, and the resulting database was 16 GB which was surprising. Any idea if there are plans to make the internal format more space efficient or backward and forwards compatible?

**Reply by wrb2** [reply]
I remember someone mentioning backwards compatibility from 1.0, but I have no idea when that might be.
